Automattic/jetpack
# Add "warm welcome" for newly purchased plans

Fixes #7374

This PR works in the following technical ways:
 - If the Jetpack plan changes, an option `show_welcome_for_new_plan` will be set and act as a flag to whether or not a "new plan" was purchased. It should not set if the plan changes/downgrades to "free"
- New `<WelcomeNewPlan />` component, which acts as a "parent" component to the dummy individual welcomes.  
- Dismissing simply sets the flag option to false.

To Test: 
- Upgrade to any plan.  
- You should see the warm welcome on any admin screen
- Dismissing should dismiss
- Links should all dismiss and take you to the respective places
- Should not be able to see it at all if not an admin, or on free plan of any kind.

Hacks to make testing easier: 
- manually set the flag option `update_option( 'show_welcome_for_new_plan', true );` and then using the dev tools, select a paid plan.
- Since the modal is hiding the "dev tools" footer link, you can manually trigger it by writing `window.dispatch( { type: "DEV_CARD_DISPLAY" } );` in the browser console.  

In progress until @MichaelArestad gets his hands on it
